### PR TITLE
User can save time

### PR DESCRIPTION
**Pull request by vick3d**
#### Description
As a User
In order to keep track of my working time
I would like to be able to save it

Changes proposed in this pull request:

* Render a time tracking view
* Add save button
* Create post requests to API
* Mock API testing
* Add styling

#### What I have learned working on this feature: 
Writing test in Cypress, Mock out API requests 

#### Screenshots:
image

### CODE REVIEW

File: cypress/integration/timeTrackingTable.feature.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+describe('Display time tracking table', () => {
+
```
Comment (LauraRe) [inline cypress/integration/timeTrackingTable.feature.js]: remove space

File: cypress/integration/timeTrackingTable.feature.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  it('when user visits the page', () => {
+		cy.visit('http://localhost:3000')
+		cy.get('div[name="timetracking"]')
+			.should('contain', 'Time Tracking')
+
```
Comment (LauraRe) [inline cypress/integration/timeTrackingTable.feature.js]: remove space

File: cypress/integration/user_can_save_time.feature.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
+describe('User can save time', () => {
+	it('User can save time', () => {
+
```
Comment (LauraRe) [inline cypress/integration/user_can_save_time.feature.js]: remove space

File: cypress/integration/user_can_save_time.feature.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+			response: {message: 'Validation Failed'},
+			headers: {
+				"X-AUTH-USER": "susan_super",
+				"X-AUTH-TOKEN": "api_kitten"
+			}
+		})
+
+		const stub = cy.stub()
+		cy.on ('window:alert', stub)
+
```
Comment (LauraRe) [inline cypress/integration/user_can_save_time.feature.js]: remove space

File: cypress/support/index.js [hunk 5]
```diff
@@ -1 +1,6 @@
-import './commands'
\ No newline at end of file
+import './commands'
+
+
```
Comment (LauraRe) [inline cypress/support/index.js]: remove space

File: src/components/timeTrackingTable.js [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+				<Table.Header name="tableHeader">
+					<Table.Row name="tableRow">
+						<Table.HeaderCell>Start Time</Table.HeaderCell>
+						<Table.HeaderCell>End Time</Table.HeaderCell>
+						<Table.HeaderCell>Rate</Table.HeaderCell>
+						<Table.HeaderCell>Customer</Table.HeaderCell>
+						<Table.HeaderCell>Project</Table.HeaderCell>
+						<Table.HeaderCell>Task</Table.HeaderCell>
+						<Table.HeaderCell> </Table.HeaderCell>
+
```
Comment (LauraRe) [inline src/components/timeTrackingTable.js]: remove space

File: src/modules/kimaiSaveTimeData.js [hunk 7]
```diff
@@ -0,0 +1,46 @@
+import axios from 'axios'
+
+
```
Comment (LauraRe) [inline src/modules/kimaiSaveTimeData.js]: remove space

File: src/modules/kimaiSaveTimeData.js [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+		axios.post(apiUrl + 'timesheets', {
+								begin: begin,
+								end: end,
+								customer: customer,
+								project: project,
+								activity: activity,
+								description: description,
+								fixedRate: fixedRate,
+								hourlyRate: hourlyRate
+						},{
```
Comment (LauraRe) [inline src/modules/kimaiSaveTimeData.js]: maybe you cab go down one line and keep } on same level as {

File: cypress/integration/timeTrackingTable.feature.js (hunk 1 above)
Comment (amandagibson) [reply cypress/integration/timeTrackingTable.feature.js]: done

File: cypress/integration/timeTrackingTable.feature.js (hunk 2 above)
Comment (amandagibson) [reply cypress/integration/timeTrackingTable.feature.js]: done

File: cypress/integration/user_can_save_time.feature.js (hunk 3 above)
Comment (amandagibson) [reply cypress/integration/user_can_save_time.feature.js]: done

File: cypress/integration/user_can_save_time.feature.js (hunk 4 above)
Comment (amandagibson) [reply cypress/integration/user_can_save_time.feature.js]: done

File: cypress/support/index.js (hunk 5 above)
Comment (amandagibson) [reply cypress/support/index.js]: done

File: src/components/timeTrackingTable.js (hunk 6 above)
Comment (amandagibson) [reply src/components/timeTrackingTable.js]: done

File: src/modules/kimaiSaveTimeData.js (hunk 7 above)
Comment (amandagibson) [reply src/modules/kimaiSaveTimeData.js]: done

File: src/modules/kimaiSaveTimeData.js (hunk 8 above)
Comment (amandagibson) [reply src/modules/kimaiSaveTimeData.js]: fixed

File: src/App.js [hunk 9]
```diff
@@ -1,16 +1,26 @@
...
 				<Navbar />
 				<LandingPage />
 				<Footer />
-      </>
+				<Segment name="timetracking">
+					<Header as='h1'textAlign='center'>
+						Time Tracking
+					</Header>
+					<TimeTrackingTable
+					/>
```
[the remark this answers is not in the document]
Comment (amandagibson) [reply src/App.js]: done

File: src/App.js [hunk 10]
```diff
@@ -1,16 +1,26 @@
 import React, { Component } from 'react';
+import { TimeTrackingTable } from './components/timeTrackingTable';
+import { Segment, Header} from 'semantic-ui-react';
 import LandingPage from './components/landingPage';
 import Navbar from './components/navbar'
 import Footer from './components/footer';
 
 class App extends Component {
+
```
[the remark this answers is not in the document]
Comment (amandagibson) [reply src/App.js]: removed

**Pull request merged**